src/draw/shapes.typ

```diff
@@ -788,47 +794,82 @@
+      let bh = (baseline-height - padding.top - padding.bottom) / 2
+
+      let bounds-center = if auto-size {
         a
       } else {
         vector.lerp(a, b, .5)
       }
 
       // Only the center anchor gets transformed. All other anchors
       // must be calculated relative to the transformed center!
-      center = matrix.mul4x4-vec3(ctx.transform,
-        vector.as-vec(center, init: (0,0,0)))
+      bounds-center = matrix.mul4x4-vec3(ctx.transform,
+        vector.as-vec(bounds-center, init: (0,0,0)))
+
+      let east-dir = vector.rotate-z((1, 0, 0), angle)
+      let north-dir = vector.rotate-z((-1, 0, 0), angle + 90deg)
+      let east-scaled = vector.scale(east-dir, +w)
+      let west-scaled = vector.scale(east-dir, -w)
+      let north-scaled = vector.scale(north-dir, +h)
+      let south-scaled = vector.scale(north-dir, -h)
+
+      let north = vector.add(bounds-center, north-scaled)
+      let south = vector.add(bounds-center, south-scaled)
+      let east = vector.add(bounds-center, east-scaled)
+      let west = vector.add(bounds-center, west-scaled)
+      let north-east = vector.add(bounds-center, vector.add(north-scaled, east-scaled))
+      let north-west = vector.sub(bounds-center, vector.add(south-scaled, east-scaled))
+      let south-east = vector.add(bounds-center, vector.add(south-scaled, east-scaled))
+      let south-west = vector.sub(bounds-center, vector.add(north-scaled, east-scaled))
+
+      let base = vector.add(south,
+        vector.scale(north-dir, padding.bottom + baseline-height))
+      let mid = vector.lerp(
+        vector.sub(north, vector.scale(north-dir, padding.top)),
+        base,
+        0.5)
+      let base-east = vector.add(base, east-scaled)
+      let base-west = vector.add(base, west-scaled)
+      let mid-east = vector.add(mid, east-scaled)
+      let mid-west = vector.add(mid, west-scaled)
 
-      let north = (calc.sin(angle)*h, -calc.cos(angle)*h,0)
-      let east = (calc.cos(-angle)*w, -calc.sin(-angle)*w,0)
-      let south = vector.scale(north, -1)
-      let west = vector.scale(east, -1)
       (
-        center: center,
-        north: vector.add(center, north),
-        north-east: vector.add(center, vector.add(north, east)),
-        east: vector.add(center, east),
-        south-east: vector.add(center, vector.add(south, east)),
-        south: vector.add(center, south),
-        south-west: vector.add(center, vector.add(south, west)),
-        west: vector.add(center, west),
-        north-west: vector.add(center, vector.add(north, west)),
+        center: bounds-center,
+        mid: mid,
+        mid-east: mid-east,
+        mid-west: mid-west,
+        base: base,
+        base-east: base-east,
+        base-west: base-west,
+        north: north,
+        north-east: north-east,
+        north-west: north-west,
+        south: south,
+        south-east: south-east,
+        south-west: south-west,
+        east: east,
+        west: west,
```

Comment:
Very cool to see this added already! As far as can see, this PR currently only adds `base-*` and `mid-*` anchors. All that remains to achieve parity with Tikz in this regard is the `text` anchor, which lies where the contents baseline and western edge meet.